### PR TITLE
Domains Dashboard Card: Card UI v2

### LINKED ISSUE #20550: Domains Dashboard Card: Card UI v2
**Specification:** pc8eDl-LC-p2

Create Domains Dashboard Card

## Requirements

1. The card should resemble the design requirements
2. The card should support light and dark mode
3. The card should support accessibility and voice-over features
4. The card should show "domain.blog" static text (Use [Domains Dashboard Card UI v2.1 SVG.zip](https://github.com/wordpress-mobile/WordPress-Android/files/11269440/Domains.Dashboard.Card.UI.v2.1.SVG.zip))

## Design

(Card) Domain Upsell – iOS Light
(Card) Domain Upsell – iOS Dark

### PR DESCRIPTION
**Pull request by staskus**
Fixes #20550

## Description

Updated the UI to use "a placeholder search bar view" indicating to users the possibility to look for the domain.

## Testing instructions

1. Run the Jetpack app
2. Log into free .com website without a domain
3. Enable Domain Dashboard Card feature flag
4. Confirm that the domain dashboard card appears
5. Switch between Light / Dark mode - confirm that UI adapts
6. Increase/Decrease text size - confirm that title and description adapts and the search view remains static as an image
7. Change from vertical to horizontal orientation - confirm that UI adapts
8. Test voiceover - confirm that title, more button, and description are correctly indicated by voiceover

## Regression Notes

1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

- [x] Wait for https://github.com/wordpress-mobile/WordPress-iOS/pull/20562 to be merge and confirm that the tests continue to be working

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Images & Videos

https://user-images.githubusercontent.com/4062343/233357083-be98dce9-39dc-4d9a-992e-5217090286e0.mp4

| Light | Dark | iPad |
|-------|-------|------|
| Simulator Screen Shot - iPhone 14 - 2023-04-20 at 14 37 19 | Simulator Screen Shot - iPhone 14 - 2023-04-20 at 14 37 11 | Simulator Screen Shot - iPad Air (5th generation) - 2023-04-20 at 14 28 13 |

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Domains/DashboardDomainsCardSearchView.swift
Comment (guarani) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Domains/DashboardDomainsCardSearchView.swift]: nitpick: typo "Vew" here

**Review by guarani — APPROVED** [review]
Thanks for this. I didn't realize the gradient was drawn natively at first (I thought it was an image). Nice work!

Tested light/dark modes, VoiceOver, and portrait/landscape on an iPhone. Also tested it still takes me to the domain flow when tapped.

Left one non-blocking comment for a typo.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Domains/DashboardDomainsCardSearchView.swift
Comment (staskus) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Domains/DashboardDomainsCardSearchView.swift]: Thanks, I'll definitely want to fix before merging!

**Comment by staskus** [comment]
> Thanks for this. I didn't realize the gradient was drawn natively at first (I thought it was an image). Nice work!

@guarani, thanks, yes, the original idea was to use an image, however, I couldn't satisfy all the requirements related to different orientations and screen sizes when using an image. In all fairness, it was quite fun to develop this small view.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Domains/DashboardDomainsCardSearchView.swift
Comment (guarani) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Domains/DashboardDomainsCardSearchView.swift]: Cool! I agree with fixing this before merging. I mostly wanted to make sure you don't need to wait for another review after fixing to merge. Also wanted to let you merge as-is if there's an urgent need to land this (e.g. if you wanted to fix the typo in a later PR).

**Pull request merged**
